### PR TITLE
Add a config name to a download url

### PR DESCRIPTION
**Pull request by vaskonov**
In order to pair the requested file to download with corresponding config file, the config name is added to the requested file url, for instance 

http://files.deeppavlov.ai/deeppavlov_data/ner_conll2003_v4_cpu_compatible.tar.gz?config=ner_conll2003

### CODE REVIEW

File: deeppavlov/download.py [hunk 1]
```diff
@@ -76,7 +76,12 @@ def get_configs_downloads(config: Optional[Union[str, Path, dict]]=None) -> Dict
 
 
 def check_md5(url: str, dest_paths: List[Path]) -> bool:
-    r = requests.get(url + '.md5')
+    if "?" in url:
+        url_md5 = url.replace("?", '.md5?')
```
Comment (yoptar) [inline deeppavlov/download.py]: You can use https://docs.python.org/3.6/library/urllib.parse.html#urllib.parse.urlparse
and https://docs.python.org/3.6/library/collections.html#collections.somenamedtuple._replace

File: deeppavlov/download.py [hunk 2]
```diff
@@ -140,6 +146,7 @@ def deep_download(config: Union[str, Path, dict]) -> None:
     downloads = get_configs_downloads(config)
 
     for url, dest_paths in downloads.items():
+        url += "?config=" + config.stem
```
Comment (yoptar) [inline deeppavlov/download.py]: what if the url already has a `?` character?

Comment (vaskonov) [reply deeppavlov/download.py]: added a def to treat this case

File: deeppavlov/download.py (hunk 1 above)
Comment (vaskonov) [reply deeppavlov/download.py]: add a def to modify an URL

File: deeppavlov/download.py [hunk 3]
```diff
@@ -140,6 +141,8 @@ def deep_download(config: Union[str, Path, dict]) -> None:
     downloads = get_configs_downloads(config)
 
     for url, dest_paths in downloads.items():
+        if isinstance(config, Path):
+            url = set_query_parameter(url, 'config', config.stem)
```
Comment (yoptar) [inline deeppavlov/download.py]: `config` can be a string
```python
        if not isinstance(config, dict):
            url = set_query_parameter(url, 'config', Path(config).stem)
```

**Pull request merged**